src/display/canvas.js

```diff
@@ -2120,17 +2122,19 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
           newHeight = Math.ceil(paintHeight / 2);
           heightScale /= paintHeight / newHeight;
         }
-        tmpCanvas = this.cachedCanvases.getCanvas(tmpCanvasId,
-                                                  newWidth, newHeight);
-        tmpCtx = tmpCanvas.context;
-        tmpCtx.clearRect(0, 0, newWidth, newHeight);
-        tmpCtx.drawImage(imgToPaint, 0, 0, paintWidth, paintHeight,
-                                     0, 0, newWidth, newHeight);
-        imgToPaint = tmpCanvas.canvas;
+
         paintWidth = newWidth;
         paintHeight = newHeight;
-        tmpCanvasId = tmpCanvasId === 'prescale1' ? 'prescale2' : 'prescale1';
       }
+
+      tmpCanvas = this.cachedCanvases.getCanvas(tmpCanvasId,
```

Comment:
I think this change makes sense, but I can also recall that painting on the temporary canvases happened for a reason. @yurydelendik Could you perhaps comment on this, i.e., is there a specific reason why we're constantly painting on temporary canvases instead of only once in the end? Does it improve quality somehow?